static/app/components/events/interfaces/spans/newTraceDetailsSpanBar.tsx

```diff
@@ -0,0 +1,1220 @@
+  spanBarColor?: string;
+  spanBarType?: SpanBarType;
+  toggleSiblingSpanGroup?: ((span: SpanType, occurrence: number) => void) | undefined;
+};
+
+export class NewTraceDetailsSpanBar extends Component<SpanBarProps> {
+  componentDidMount() {
+    this._mounted = true;
+    this.updateHighlightedState();
+
+    if (this.isHighlighted && this.props.onRowClick) {
+      this.props.onRowClick(this.getSpanDetailsProps());
+    }
+
+    if (this.spanRowDOMRef.current) {
+      this.props.storeSpanBar(this);
+      this.connectObservers();
+    }
+
+    if (this.spanTitleRef.current) {
+      this.spanTitleRef.current.addEventListener('wheel', this.handleWheel, {
+        passive: false,
+      });
+    }
+
+    // On mount, it is necessary to set the left styling of the content here due to the span tree being virtualized.
+    // If we rely on the scrollBarManager to set the styling, it happens too late and awkwardly applies an animation.
+    if (this.spanContentRef) {
+      this.props.addContentSpanBarRef(this.spanContentRef);
+      const left = -this.props.getCurrentLeftPos();
+      this.spanContentRef.style.transform = `translateX(${left}px)`;
+      this.spanContentRef.style.transformOrigin = 'left';
+    }
+
+    const {
+      span,
+      markAnchoredSpanIsMounted,
+      addExpandedSpan,
+      isSpanExpanded,
+      measure,
+      didAnchoredSpanMount,
+    } = this.props;
+
+    if (isGapSpan(span)) {
+      return;
+    }
+
+    if (spanTargetHash(span.span_id) === location.hash && !didAnchoredSpanMount()) {
+      this.scrollIntoView();
+      markAnchoredSpanIsMounted?.();
+      addExpandedSpan(span);
+      return;
+    }
+
+    if (isSpanExpanded(span)) {
+      this.setState({showDetail: true}, measure);
+    }
+  }
+
+  componentDidUpdate(prevProps: Readonly<SpanBarProps>): void {
+    if (this.props.location.query !== prevProps.location.query) {
+      this.updateHighlightedState();
+    }
+  }
+
+  componentWillUnmount() {
+    this._mounted = false;
+    this.disconnectObservers();
+
+    if (this.spanTitleRef.current) {
+      this.spanTitleRef.current.removeEventListener('wheel', this.handleWheel);
+    }
+
+    const {span} = this.props;
+    if (isGapSpan(span)) {
+      return;
+    }
+
+    this.props.removeContentSpanBarRef(this.spanContentRef);
+  }
+
+  spanRowDOMRef = createRef<HTMLDivElement>();
+  spanTitleRef = createRef<HTMLDivElement>();
+
+  spanContentRef: HTMLDivElement | null = null;
+  intersectionObserver?: IntersectionObserver = void 0;
+  zoomLevel: number = 1; // assume initial zoomLevel is 100%
+  _mounted: boolean = false;
+  span_id: string | string[] | null | undefined = null;
+  isHighlighted: boolean = false;
+
+  updateHighlightedState = () => {
+    this.span_id = this.props.location.query.span;
+    this.isHighlighted = !!(
+      !isGapSpan(this.props.span) &&
+      this.span_id &&
+      this.span_id === this.props.span.span_id
+    );
+    this.forceUpdate();
+  };
```

Comment:
hmm is `forceUpdate` necessary here? I feel like the component will rerender anyways because the location changes

Reply:
Yeah it is necessary.  Without it I encountered this weird highlight on double click issue ,where the first click highlights the previous bar clicked. I can solve this by converting it to a functional component but this was just too big of a component to got at at the moment. Will add a todo for this.